includes/class-wc-admin-api-init.php

```diff
@@ -312,39 +373,52 @@ public static function orders_data_store_init() {
 	}
 
 	/**
-	 * Init orders product lookup store.
-	 *
-	 * @param WC_Background_Updater|null $updater Updater instance.
-	 * @return bool
+	 * Init order/product lookup tables update (in batches).
 	 */
-	public static function order_product_lookup_store_init( $updater = null ) {
-		// TODO: this needs to be updated a bit, as it no longer runs as a part of WC_Install, there is no bg updater.
-		global $wpdb;
+	public static function orders_lookup_batch_init() {
+		$batch_size  = self::get_batch_size( self::ORDERS_BATCH_ACTION );
+		$order_query = new WC_Order_Query(
+			array(
+				'return'   => 'ids',
+				'limit'    => 1,
+				'paginate' => true,
+			)
+		);
+		$result      = $order_query->get_orders();
 
-		$orders = get_transient( 'wc_update_350_all_orders' );
-		if ( false === $orders ) {
-			$orders = wc_get_orders(
-				array(
-					'limit'  => -1,
-					'return' => 'ids',
-				)
-			);
-			set_transient( 'wc_update_350_all_orders', $orders, DAY_IN_SECONDS );
+		if ( 0 === $result->total ) {
+			return;
 		}
 
-		// Process orders until close to running out of memory timeouts on large sites then requeue.
-		foreach ( $orders as $order_id ) {
+		$num_batches = ceil( $result->total / $batch_size );
+
+		self::queue_batches( 1, $num_batches, self::ORDERS_BATCH_ACTION );
+	}
+
+	/**
+	 * Process a batch of orders to update (stats and products).
+	 *
+	 * @param int $batch_number Batch number to process (essentially a query page number).
+	 * @return void
+	 */
+	public static function orders_lookup_process_batch( $batch_number ) {
+		$batch_size  = self::get_batch_size( self::ORDERS_BATCH_ACTION );
+		$order_query = new WC_Order_Query(
+			array(
+				'return'  => 'ids',
+				'limit'   => $batch_size,
+				'page'    => $batch_number,
+				'orderby' => 'ID',
+				'order'   => 'ASC',
+			)
+		);
```

Comment:
Will this include refunds correctly?

Reply:
I think you're pointing out the difference in arguments vs. the previous `wc_get_orders()` call ([#](https://github.com/woocommerce/wc-admin/pull/1291/files#diff-86dddbc0241abd7346998f87a6b24228L385))- right?

If so - I think this **does** include refunds, which is a departure from the original logic, and probably a bug.